allennlp/data/token_indexers/token_indexer.py

```diff
@@ -96,6 +100,14 @@ def pad_token_sequence(self,
         """
         raise NotImplementedError
 
+    def array_type(self) -> Type[torch.Tensor]: # pylint: disable=no-self-use
```

Comment:
why this (indirect) design rather than something more explicit like

```
def make_tensor(self, pre_tensor) -> torch.Tensor:
    """override if you don't want long tensors"""
    return torch.LongTensor(pre_tensor)
```

?

Reply:
I was wondering about all of the separate operations done in the interaction between `TextField` and `TokenIndexer`, also.  We could just change `pad_token_sequence` into an `as_padded_tensor` function, or something, and there may be other places we can similarly simplify.  That's probably for a separate PR, though.  Except that we don't really want to introduce a function here that we're going to remove later...